compiler/src/dotty/tools/dotc/core/tasty/TastyFormat.scala

```diff
@@ -231,38 +224,48 @@ Standard Section: "Positions" Assoc*
 object TastyFormat {
 
   final val header = Array(0x5C, 0xA1, 0xAB, 0x1F)
-  val MajorVersion = 1
-  val MinorVersion = 1
-
-  // Name tags
-
-  final val UTF8 = 1
-  final val QUALIFIED = 2
-  final val FLATTENED = 3
-  final val EXPANDED = 4
-  final val EXPANDPREFIX = 5
-  final val TRAITSETTER = 6
-  final val UNIQUE = 10
-  final val DEFAULTGETTER = 11
-  final val VARIANT = 12
-  final val OUTERSELECT = 13
-
-  final val SUPERACCESSOR = 20
-  final val PROTECTEDACCESSOR = 21
-  final val PROTECTEDSETTER = 22
-  final val INITIALIZER = 23
-  final val AVOIDCLASH = 30
-  final val DIRECT = 31
-  final val FIELD = 32
-  final val EXTMETH = 33
-  final val ADAPTEDCLOSURE = 34
-  final val OBJECTVAR = 39
-  final val OBJECTCLASS = 40
-
-  final val SIGNED = 63
-
-  final val firstInternalTag = 64
-  final val IMPLMETH = 64
+  val MajorVersion = 2
+  val MinorVersion = 0
+
+  /** Tags used to serialize names */
+  class NameTags {
+    final val UTF8 = 1               // A simple name in UTF8 encoding.
+
+    final val QUALIFIED = 2          // A fully qualified name `<prefix>.<suffix>`.
+
+    final val EXPANDED = 3           // An expanded name `<prefix>$$<suffix>`,
+                                     // used by Scala-2 for private names.
+
+    final val EXPANDPREFIX = 4       // An expansion prefix `<prefix>$<suffix>`,
+                                     // used by Scala-2 for private names.
+
+    final val UNIQUE = 10            // A unique name `<name>$<num>` where `<num>`
```

Comment:
Since we're breaking compat anyway, we could renumber the tags so that all the tasty tags are continuous (and non-tasty nametags could have numbers starting with 1000 to distinguish them)

Reply:
I tried this but got weird bootstrap errors and don't have the time to chase them down. Maybe somebody else wants to give it a try? Since so much else is sitting on this commit, I am getting it merged now.